### PR TITLE
Add nifti_to_zarr_ngff

### PR DESCRIPTION
**Pull request by thewtex**

### CODE REVIEW

**Comment by thewtex** [comment]
There is a confusing warning:

```
miqa/core/conversion/nifti_to_zarr_ngff.py:4:1: I001 isort found an import in
the wrong position
from pathlib import Path
```

Any ideas?

**Comment by dzenanz** [comment]
`pathlib` sorts before `typing`:
```py
from pathlib import Path
from typing import Union
```
I do not have the right to push to your fork.

File: miqa/core/tests/test_zarr.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        / Path(
+            'datasnap-2019-01-23/fs/storage/XNAT/archive/ohsu_incoming/arc001/D-99999-P-9-2081220/RESOURCES/nifti/2_ncanda-mprage-v1/image.nii.gz'  # noqa: E501
+        ),
+        samples_dir
+        / Path(
+            'datasnap-2019-01-23/fs/storage/XNAT/archive/ohsu_incoming/arc001/D-99999-P-9-2081220/RESOURCES/nifti/2_ncanda-mprage-v1/image.nii.gz'  # noqa: E501
+        ),
+    ]
+
+    sample = samples[0]
```
Comment (dzenanz) [inline miqa/core/tests/test_zarr.py:22]: unused variable

**Comment by dzenanz** [comment]
Please rebase on current master. What is the right way to invoke the supplied test?

**Comment by thewtex** [comment]
> pathlib sorts before typing:

Ah, I was looking in the wrong file -- thanks!

**Comment by thewtex** [comment]
> Please rebase on current master. What is the right way to invoke the supplied test?

The test can be run with `tox`

**Comment by thewtex** [comment]
> Please rebase on current master.

I rebased on `master`. After the rebase, the tests are failing locally with:

```
ROR: InvocationError for command /home/matt/src/miqa/.tox/test/bin/pytest (exited with code 1)
check-migrations inst-nodeps: /home/matt/src/miqa/.tox/.tmp/package/1/miqa-0.1.0.zip
check-migrations installed: amqp==5.0.6,asciitree==0.3.3,asgiref==3.4.1,attrs==21.2.0,backcall==0.2.0,backports.entry-points-selectable==1.1.0,billiard==3.6.4.0,boto3==1.18.38,botocore==1.21.38,Brotli==1.0.9,celery==5.1.2,certifi==2021.5.30,cffi==1.14.6,charset-normalizer==2.0.4,click==7.1.2,click-didyoumean==0.0.3,click-plugins==1.1.1,click-repl==0.2.0,colorama==0.4.4,commonmark==0.9.1,configparser==5.0.2,coreapi==2.3.3,coreschema==0.0.4,cryptography==3.4.8,decorator==5.0.9,defusedxml==0.7.1,Deprecated==1.2.13,distlib==0.3.2,dj-database-url==0.5.0,dj-email-url==1.0.2,Django==3.2.7,django-allauth==0.45.0,django-click==2.3.0,django-composed-configuration==0.17.0,django-configurations==2.2,django-cors-headers==3.8.0,django-debug-toolbar==3.2.2,django-extensions==3.1.3,django-filter==2.4.0,django-girder-style==0.6.0,django-girder-utils==0.8.0,django-minio-storage==0.3.10,django-oauth-toolkit==1.5.0,django-s3-file-field==0.1.2,django-storages==1.11.1,djangorestframework==3.12.4,drf-yasg==1.20.0,factory-boy==3.2.0,Faker==8.12.1,fasteners==0.16.3,filelock==3.0.12,gunicorn==20.1.0,idna==3.2,inflection==0.5.1,ipython==7.27.0,itk-core==5.2.1.post1,itk-filtering==5.2.1.post1,itk-io==5.2.1.post1,itk-numerics==5.2.1.post1,itypes==1.2.0,jedi==0.18.0,Jinja2==3.0.1,jmespath==0.10.0,jsonschema==3.2.0,jwcrypto==1.0,kombu==5.1.0,MarkupSafe==2.0.1,matplotlib-inline==0.1.3,minio==6.0.2,miqa @ file:///home/matt/src/miqa/.tox/.tmp/package/1/miqa-0.1.0.zip,numcodecs==0.9.1,numpy==1.21.2,oauthlib==3.1.1,packaging==21.0,pandas==1.3.2,parso==0.8.2,pexpect==4.8.0,pickleshare==0.7.5,platformdirs==2.3.0,pluggy==1.0.0,prompt-toolkit==3.0.20,psycopg2==2.9.1,ptyprocess==0.7.0,py==1.10.0,pycparser==2.20,Pygments==2.10.0,PyJWT==2.1.0,pyparsing==2.4.7,pyrsistent==0.18.0,python-dateutil==2.8.2,python3-openid==3.2.0,pytz==2021.1,requests==2.26.0,requests-oauthlib==1.3.0,rich==10.9.0,ruamel.yaml==0.17.16,ruamel.yaml.clib==0.2.6,s3transfer==0.5.0,sentry-sdk==1.3.1,six==1.16.0,spatial_image==0.0.3,spatial_image_multiscale==0.2.0,spatial_image_ngff==0.1.0,sqlparse==0.4.1,text-unidecode==1.3,toml==0.10.2,tox==3.24.3,traitlets==5.1.0,uritemplate==3.0.1,urllib3==1.26.6,vine==5.0.0,virtualenv==20.7.2,wcwidth==0.2.5,whitenoise==5.3.0,wrapt==1.12.1,xarray==0.19.0,zarr==2.9.5
check-migrations run-test-pre: PYTHONHASHSEED='3762926365'
check-migrations run-test: commands[0] | /home/matt/src/miqa/.tox/check-migrations/bin/python ./manage.py makemigrations --check --dry-run
Traceback (most recent call last):
  File "/home/matt/src/miqa/./manage.py", line 20, in <module>
    main()
  File "/home/matt/src/miqa/./manage.py", line 16, in main
    execute_from_command_line(sys.argv)
  File "/home/matt/src/miqa/.tox/check-migrations/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/home/matt/src/miqa/.tox/check-migrations/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
    django.setup()
  File "/home/matt/src/miqa/.tox/check-migrations/lib/python3.9/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/matt/src/miqa/.tox/check-migrations/lib/python3.9/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/home/matt/src/miqa/.tox/check-migrations/lib/python3.9/site-packages/django/apps/config.py", line 301, in import_models
    self.models_module = import_module(models_module_name)
  File "/home/matt/bin/mambaforge/envs/miqa/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/matt/src/miqa/miqa/core/models/__init__.py", line 5, in <module>
    from .project import Project
  File "/home/matt/src/miqa/miqa/core/models/project.py", line 9, in <module>
    from miqa.learning.evaluation_models import available_evaluation_models
  File "/home/matt/src/miqa/miqa/learning/evaluation_models.py", line 4, in <module>
    from uri import URI
ModuleNotFoundError: No module named 'uri'
ERROR: InvocationError for command /home/matt/src/miqa/.tox/check-migrations/bin/python ./manage.py makemigrations --check --dry-run (exited with code 1)

```

**Comment by dzenanz** [comment]
```text
File "/home/matt/src/miqa/miqa/learning/evaluation_models.py", line 4, in <module>
    from uri import URI
ModuleNotFoundError: No module named 'uri'
```
I guess we need to add `uri` to the list of [requirements](https://github.com/OpenImaging/miqa/blob/2bc3384df9a3ea14824768096c9e310fd25deb38/setup.py#L38), which you should also install locally.

**Review by annehaley — APPROVED** [review]

**Comment by dzenanz** [comment]
@annehaley Do I need to do everything [described in the readme](https://github.com/OpenImaging/miqa/blob/master/README.md#develop-natively-advanced) in order to run tests? When I invoke `tox`, I run into a this:
```text
...
check-migrations run-test-pre: PYTHONHASHSEED='653'
check-migrations run-test: commands[0] | 'M:\MIQA\miqa\.tox\check-migrations\Scripts\python' ./manage.py makemigrations --check --dry-run
Traceback (most recent call last):
  File "M:\MIQA\miqa\manage.py", line 20, in <module>
    main()
  File "M:\MIQA\miqa\manage.py", line 16, in main
    execute_from_command_line(sys.argv)
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\django\core\management\__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\django\core\management\__init__.py", line 363, in execute
    settings.INSTALLED_APPS
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\django\conf\__init__.py", line 82, in __getattr__
    self._setup(name)
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\django\conf\__init__.py", line 69, in _setup
    self._wrapped = Settings(settings_module)
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\django\conf\__init__.py", line 170, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "C:\Program Files\Python39\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\configurations\importer.py", line 182, in load_module
    reraise(err, "Couldn't setup configuration '{0}'".format(cls_path))
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\configurations\importer.py", line 164, in load_module
    cls.setup()
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\configurations\base.py", line 127, in setup
    setup_value(cls, name, value)
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\configurations\values.py", line 15, in setup_value
    actual_value = value.setup(name)
  File "M:\MIQA\miqa\.tox\check-migrations\lib\site-packages\configurations\values.py", line 105, in setup
    raise ValueError('Value {0!r} is required to be set as the '
ValueError: Couldn't setup configuration 'miqa.settings.TestingConfiguration':  Value 'DATABASES' is required to be set as the environment variable 'DJANGO_DATABASE_URL'
ERROR: InvocationError for command 'M:\MIQA\miqa\.tox\check-migrations\Scripts\python' ./manage.py makemigrations --check --dry-run (exited with code 1)
___________________________________________________________________________ summary ___________________________________________________________________________
  lint: commands succeeded
ERROR:   test: commands failed
ERROR:   check-migrations: commands failed
```

**Comment by annehaley** [comment]
hm, have you done `source dev/export-env.sh`? I remember @dchiquito told me that was a precursor to using tox. If that's the issue, maybe that should also be included in the README.

**Comment by dzenanz** [comment]
No, because I am developing on Windows.

**Comment by annehaley** [comment]
Ah, we may have to ask @dchiquito if there is a windows equivalent procedure.

**Comment by dchiquito** [comment]
`dev/export-env.sh` is just a wrapper to get all the variables in `dev/.env.docker-compose-native` into your local environment. Before we had that script I wrote my own .sh script that hardcoded all those values and exported them.

I've never encountered anyone with this problem before, so I assume no one else in D&A uses windows for development.

**Pull request merged**